modules/openapi-generator/src/main/java/org/openapitools/codegen/utils/ModelUtils.java

```diff
@@ -909,15 +911,19 @@ public static String getParentName(ComposedSchema composedSchema, Map<String, Sc
                         // discriminator.propertyName is used
                         return parentName;
                     } else {
-                        LOGGER.debug("Not a parent since discriminator.propertyName is not set {}", s.get$ref());
-                        // not a parent since discriminator.propertyName is not set
+                        refedParentNames.add(parentName);
                     }
                 } else {
                     // not a ref, doing nothing
                 }
             }
         }
 
+        // parent name only makes sense when there is a single obvious parent
+        if (refedParentNames.size() == 1) {
+            return refedParentNames.get(0);
```

Comment:
Can we throw a warning about this behavior and advise the user to use "discriminator.propertyName" in OAS v3 instead to set a proper parent model as such behavior will be removed in the future?

Our goal is to move users to adopt OAS v3 instead of staying with OAS v2 (which is not the best specification to work with inheritance/composition)

Reply:
Sure, I'd still be interested in how you would recommend applying the discriminator approach to a collection of schema's like I describe in the MR description - the parent in this case should not have knowledge of its children, and also I don't particularly want to add an arbitrary field like `type: string` solely for code generation purposes.

I'd be more than happy to refactor our definitions to do things the OAS v3 way over time - in general I've found v3 much better to work with than v2 ever was.

Reply:
we should raise an issue at the openapi 3.0 specification to allow this type of inheritance without discriminator

Reply:
Another way is to use vendor extension to describe this if the OAS 3.0 spec doesn't officially support it at the moment.

(nullable was not supported in OAS v2 and we'd to use x-nullable: true before OAS v3 officially supports it)

Reply:
this is a great idea, and probably the easiest way to support this behavior in the future.

Reply:
That sounds like a good plan. @macjohnny would you be comfortable raising such an issue? I don't really have the bandwidth to write a well thought proposal at this time.

Also I've added a warning, and fixed the commit author (thanks for pointing that out). Let me know if you're happy with the wording

Reply:
Btw, if this fallback is breaking some existing use cases (allOf with just one schema), we will need to add an option later to cater to both cases.

Reply:
@wing328 in regards to your https://github.com/OpenAPITools/openapi-generator/pull/3771#discussion_r318663859 and mine  https://github.com/OpenAPITools/openapi-generator/pull/3771#issuecomment-538341900 I assume that's what is happening in my case.